### DISCUSSION TITLE
GeoIP2 Configuration - Documentation

### DISCUSSION THREAD

**Discussion by dougc84** [dokku/dokku#5680] [Q&A]
### Description of feature

I've been using Dokku recently and I love it! It's been a great staging environment and a great experience overall, and I'm planning on using Dokku for all future deploys.

However, I want to use the GeoIP2 module with nginx to block users by IP. I've worked this out with my Rails app using a combination of a dokku plugin (to download and maintain the MaxMind database) and using `rack-attack`, but I'd rather block users before they ever hit my app. It makes for less false errors in error reporting software (you know, people trying to get at your `/.git/credentials` file or something), and doesn't require the load of hitting a Rails app and serving up a response.

I've tried multiple ways of configuring nginx - using sigil templates and such - but they always fail. It looks like the geoip2 module is installed by default, but it maybe isn't loaded? So I'm not really sure how to make this function properly. I've done this other times with capistrano deploys and... yeah, I don't want to go back to those days!

So the feature request is that I'd love to see some documentation on how to load modules with nginx, as well as some examples. Or, if that's not possible, to add this functionality.

Thanks so much!

**Comment by josegonzalez** [comment]
> I've tried multiple ways of configuring nginx - using sigil templates and such - but they always fail. 

What do you mean by fail?

**Comment by dougc84** [comment]
@josegonzalez I thought I had mentioned this above - a lot of “this declaration doesn’t belong here” messages despite thats where configuration is supposed to go.

**Comment by josegonzalez** [comment]
Can you show me what you tried?
